src/frontends/tensorflow/src/tf_utils.cpp

```diff
@@ -542,23 +551,46 @@ shared_ptr<v5::Loop> create_loop_for_tf_while(const std::string& while_node_name
     // set data for the Loop node
     loop->set_function(body_model);
 
+    std::vector<size_t> invariant_input_indexes;
     // body_results may contain less nodes than body_params that means back edge exists not for all body_params
     for (size_t input_ind = 0; input_ind < body_condition_output_idx; ++input_ind) {
+        if (get_parent_parameter(body_results[input_ind])) {
+            invariant_input_indexes.push_back(input_ind);
+            continue;
+        }
         loop->set_merged_input(body_params[input_ind], ov_inputs[input_ind], body_results[input_ind]->input_value(0));
     }
+    for (size_t idx : invariant_input_indexes) {
+        loop->set_invariant_input(body_params[idx], ov_inputs[idx]);
+    }
     loop->set_special_body_ports({-1, static_cast<int64_t>(body_condition_output_idx)});
     // set invariant inputs for the loop
     for (size_t input_ind = body_condition_output_idx; input_ind < input_size; ++input_ind) {
         loop->set_invariant_input(body_params[input_ind], ov_inputs[input_ind]);
     }
 
+    std::unordered_map<size_t, Output<Node>> loop_input_nodes;  // map Parameter instance_id to input node
+    {
+        for (const auto& description : loop->get_input_descriptions()) {
+            const size_t body_input_id = body_params[description->m_body_parameter_index]->get_instance_id();
+            loop_input_nodes[body_input_id] = loop->input_value(description->m_input_index);
+        }
+    }
+    ov::OutputVector loop_outputs;
     // set external outputs for Loop node
     // do not get execution condition outside of the Loop node
     for (size_t output_ind = 0; output_ind < body_condition_output_idx; ++output_ind) {
-        loop->get_iter_value(body_results[output_ind]);
+        auto parent_parameter = get_parent_parameter(body_results[output_ind]);
+        if (parent_parameter) {
+            loop_outputs.push_back(loop_input_nodes[parent_parameter->get_instance_id()]);
+        } else {
+            const auto loop_output = loop->get_iter_value(body_results[output_ind]);
+            loop_outputs.push_back(loop_output);
+        }
     }
+
     loop->validate_and_infer_types();
-    return loop;
+    return std::make_pair(loop_outputs, loop);
```

Comment:
please return only `loop_outputs`

Reply:
In current (this PR) implementation we actually don't know if loop_output is output of the Loop or output of invariant input node. The simplest way is just to return loop with it's outputs. We can also try to find through outputs if any is loop output. @rkazants what is your opinion on this problem?

Reply:
@rkazants, is it ok if I will return tuple but not the pair?